src/mast/mod.rs

```diff
@@ -1009,13 +1064,12 @@ impl<B: Backend> Mast<B> {
                 let sig_typed = FnSig {
                     name: sig.name.clone(),
                     kind: sig.kind.clone(),
-                    generics: HashSet::new(),
+                    generics,
```

Comment:
this seem to have no effect due to not mutating generics

Reply:
I think you would have caught that if you had returned an error earlier if all the generics are not instantiated at this point

Reply:
looks like it will need to have a check to see if all the generics are instantiated before this.